dev/how-to/maintain/backup-and-restore/br.md

```diff
@@ -0,0 +1,223 @@
+---
+title: 使用 BR 进行备份与恢复
+summary: 了解如何使用 BR 工具进行集群数据备份和恢复。
+category: reference
+---
+
+# 使用 BR 进行备份与恢复
+
+Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行工具，用于对 TiDB 集群进行数据备份和恢复。相比 Mydumper，BR 更适合大数据量的场景。本文档介绍了 BR 的工作原理、命令行描述以及详细的备份恢复用例。
+
+## 工作原理
+
+BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发到各个 TiKV 节点，TiKV 收到命令后执行相应的备份和恢复操作。在一次备份或恢复中，各个 TiKV 节点都会有一个项目的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
+
+![br-arch](/media/br-arch.png)
+
+## BR 命令行描述
+
+`br` 命令行是由命令（包括子命令）、选项和参数组成的。命令即不带 `-` 或者 `--` 的字符。选项即带有 `-` 或者 `--` 的字符。参数即命令或选项字符后紧跟的、传递给命令和选项的字符。
+
+以下是一条完整的 `br` 命令行：
+
+`br backup full --pd "${PDIP}:2379" -s "local:///tmp/backup"`
+
+命令行各部分的解释如下：
+
+* `backup`：命令
+* `full`：`backup` 的子命令
+* `-s` 或 `--storage`：备份保存的路径
+* `"local:///tmp/backup"`：`-s` 的参数，保存的路径为本地磁盘的 `/tmp/backup`
+* `--pd`：PD 服务地址
+* `"${PDIP}:2379"`：`--pd` 的参数
+
+### 命令和子命令
+
+BR 由多层命令组成。目前，BR 包含 `backup`、`restore` 和 `version` 三个命令。
+
+* `br backup` 用于备份 TiDB 集群
+* `br restore` 用于恢复 TiDB 集群
+* `br version` 用于查看 BR 工具版本信息
+
+BR 还包含以下三个子命令：
+
+* `full`：可用于备份或恢复全部数据。
+* `db`：可用于备份或恢复集群中的指定数据库。
+* `table`：可用于备份或恢复集群指定数据库中的单张表。
+
+### 常用选项
+
+* `--pd`：用于连接的选项，表示 PD 服务地址，例如 `"${PDIP}:2379"`。
+* `-h`/`--help`：获取所有命令和子命令的使用帮助。例如 `br backup --help`。
+* `--ca`：指定 PEM 格式的受信任 CA 的证书文件路径。
+* `--cert`：指定 PEM 格式的 SSL 证书文件路径。
+* `--key`：指定 PEM 格式的 SSL 证书密钥文件路径。
+* `--status-addr`：指定 BR 提供 Prometheus 统计的监听地址。
+
+## 备份集群数据
+
+### 备份全部集群数据
+
+要备份全部集群数据，可使用 `br backup full` 命令。该命令的使用帮助可以通过 `br backup full -h` 或 `br backup full --help` 来获取。
+
+用例：将所有集群数据备份到各个 TiKV 节点的 `/tmp/backup` 路径
+
+{{< copyable "shell-regular" >}}
+
+```shell
+br backup full \
+    --pd ${PDIP}:2379 \
+    --storage "local:///tmp/backup" \
+    --ratelimit 120 \
+    --concurrency 4 \
+    --log-file backupfull.log
+```
+
+以上命令通过 `--ratelimit` 和 `--concurrency` 选项限制了 **每个 TiKV** 执行备份任务的并发数上限和速度上限，同时把 BR 的 log 写到 `backupfull.log` 文件中。
+
+备份期间有进度条在终端中显示。当进度条前进到 100% 时，说明备份已完成。在完成备份后，BR 为了确保数据安全性，还会校验备份数据。进度条效果如下：
+
+```shell
+br backup full \
+    --pd ${PDIP}:2379 \
+    --storage "local:///tmp/backup" \
+    --ratelimit 120 \
+    --concurrency 4 \
+    --log-file backupfull.log
+Full Backup <---------/................................................> 17.12%.
+```
+
+### 备份单张表的数据
+
+要备份集群中指定单张表的数据，可使用 `br backup table` 命令。同样可通过 `br backup table -h` 或 `br backup table --help` 来获取子命令 `table` 的使用帮助。
+
+用例：将表 `test.usertable` 备份到各个 TiKV 节点的 `/tmp/backup` 路径
+
+{{< copyable "shell-regular" >}}
+
+```shell
+br backup table \
+    --pd ${PDIP}:2379 \
+    --db test \
+    --table usertable \
+    --storage "local:///tmp/backup" \
+    --ratelimit 120 \
+    --concurrency 4 \
+    --log-file backuptable.log
+```
+
+`table` 子命令有 `--db` 和 `--table` 两个选项，分别用来指定数据库名和表名，其余选项含义一致。
+
+备份期间有进度条在终端中显示。当进度条前进到 100% 时，说明备份已完成。在完成备份后，BR 为了确保数据安全性，还会校验备份数据。
+
+## 恢复集群数据
+
+### 恢复全部备份数据
+
+要将全部备份数据恢复到集群中来，可使用 `br restore full` 命令。该命令的使用帮助可以通过 `br restore full -h` 或 `br restore full --help` 来获取。
+
+用例：将 `/tmp/backup` 路径中的全部备份数据恢复到集群中
```

Comment:
Suggest making it clear that we need to collect all the backup files and metadata under /tmp/backup dir.